docs/proposals/20210222-azure-machinepool-machine.md

```diff
@@ -0,0 +1,332 @@
+- [Proposal](#proposal)
+    - [User Stories](#user-stories)
+        - [Story 1 - Upgrading the Kubernetes Version of a MachinePool](#story-1---upgrading-the-kubernetes-version-of-a-machinepool)
+        - [Story 2 - Reducing the Number of Replicas in a MachinePool](#story-2---reducing-the-number-of-replicas-in-a-machinepool)
+    - [Requirements](#requirements)
+        - [Functional](#functional)
+        - [Non-Functional](#non-functional)
+    - [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+        - [Proposed API Changes](#proposed-api-changes)
+        - [Proposed Controllers Changes](#proposed-controller-changes)
+        - [Proposed Changes of Responsibily](#proposed-changes-of-responsibility)
+- [Available Options](#available-options-for-cluster-api-provider-azure)
+    - [Add Annotations to AzureMachinePool for Instance Delete Selection](#option-1-add-annotations-to-azuremachinepool-for-instance-delete-selection)
+        - [Pros](#option-1-pros)
+        - [Cons](#option-1-cons)
+    - [Separate AzureMachinePool and AzureMachinePoolMachines](#option-2-separate-azuremachinepool-and-azuremachinepoolmachines)
+        - [Pros](#option-2-pros)
+        - [Cons](#option-2-cons)
+- [Conclusions](#conclusions)
+- [Additional Details](#additional-details)
+    - [Test Plan](#test-plan)
+- [Implementation History](#implementation-history)    
+
+## Summary
+
+Azure MachinePool currently embeds the state of each of the machines in the MachinePool within the status of the Azure
+MachinePool. MachinePool machines should be their own resources to enable individual lifecycles.
+
+## Motivation
+
+By giving each AzureMachinePoolMachine an individual lifecycle, a user would be able to inform CAPZ of the specific
+instance to delete and then have the AzureMachinePoolMachine controller cordon and drain the node prior to deleting
+the underlying infrastructure.
+
+Azure Virtual Machine Scale Set Instances do not provide the specific machine events needed to alert Kubernetes when
+an instance is being deleted due to scaling down or being upgraded. Without some way to inform Kubernetes of the
+specific instance that is to be deleted / updated, the underlying infrastructure may be removed before workloads can be
+safely migrated from the machine / node. By managing the lifecycle from CAPZ, we are able to safely delete / upgrade
+machines / nodes.
+
+### Goals
+- Delete specific AzureMachinePool machines
+- Rolling updates with max unavailable and max surge
+  - MaxUnavailable is the max number of machines that are allowed to be unavailable at any time
+  - MaxSurge is the number of machines to surge, add to the current replica count, during an upgrade to 
+- Safely update by cordoning and draining nodes prior to deleting the underlying infrastructure
+
+### Non-Goals / Future Work
+- Create a CAPI Machine owner for each AzureMachinePoolMachine (possible future work)
+
+## Proposal
+
+### User Stories
```

Comment:
I'd love to see a user story for deleting a specific instance in a scale set

Reply:
Added.